pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -470,7 +470,7 @@ protected boolean consumeLoop()
 
       StreamPartitionMsgOffset batchFirstOffset = messageBatch.getFirstMessageOffset();
       if (batchFirstOffset != null) {
-        validateStartOffset(_currentOffset, batchFirstOffset);
+        validateStartOffset(messageBatch);
```

Comment:
No need to fetch the first message offset. We can directly check data loss from the message batch